tests/integration/targets/aws_api_gateway_domain/tasks/main.yml

```diff
@@ -0,0 +1,124 @@
+---
+# tasks file for aws_api_gateway_domain
+
+- name: Run aws_api_gateway_domain module integration tests
+  # NOTE: To make tests work set TLS ARN in defaults/main.yml to an existing and
+  # validated TLS cert on ACM that matches given domain.
+
+  block:
+
+    # ==================== preparations ========================================
+
+    - name: Preperaation - Set connection information for all tasks
+      set_fact:
+        aws_connection_info: &aws_connection_info
+          aws_access_key: "{{ aws_access_key }}"
+          aws_secret_key: "{{ aws_secret_key }}"
+          security_token: "{{ security_token }}"
+          region: "{{ aws_region }}"
+      no_log: yes
```

Comment:
No need to set the credentials and use the anchor. Please add your module here https://github.com/ansible-collections/community.aws/blob/main/meta/runtime.yml and use module_defaults instead.

``` 
module_defaults:
    group/aws:
      aws_access_key: '{{ aws_access_key }}'
      aws_secret_key: '{{ aws_secret_key }}'
      security_token: '{{ security_token | default(omit) }}'
      region: '{{ aws_region }}'
```